cli/azd/pkg/project/scaffold_gen_environment_variables.go

```diff
@@ -0,0 +1,840 @@
+					{
+						Name:  "spring.cloud.azure.eventhubs.processor.checkpoint-store.account-name",
+						Value: "${storageAccountName}",
+					},
+					{
+						Name:  "spring.cloud.azure.eventhubs.processor.checkpoint-store.credential.managed-identity-enabled",
+						Value: "true",
+					},
+					{
+						Name:  "spring.cloud.azure.eventhubs.processor.checkpoint-store.credential.client-id",
+						Value: "__PlaceHolderForServiceIdentityClientId",
+					},
+					{
+						Name:  "spring.cloud.azure.eventhubs.processor.checkpoint-store.connection-string",
+						Value: "",
+					},
+				},
+			}, nil
+		case internal.AuthTypeConnectionString:
+			return scaffold.EnvironmentVariableInformation{
+				StringEnvironmentVariables: []scaffold.StringEnvironmentVariable{
+					{
+						Name:  "spring.cloud.azure.eventhubs.processor.checkpoint-store.account-name",
+						Value: "${storageAccountName}",
+					},
+					{
+						Name:  "spring.cloud.azure.eventhubs.processor.checkpoint-store.credential.managed-identity-enabled",
+						Value: "false",
+					},
+					{
+						Name:  "spring.cloud.azure.eventhubs.processor.checkpoint-store.credential.client-id",
+						Value: "",
+					},
+				},
+				SecretRefEnvironmentVariables: []scaffold.SecretRefEnvironmentVariable{
+					{
+						Name:      "spring.cloud.azure.eventhubs.processor.checkpoint-store.connection-string",
+						SecretRef: "storage-account-connection-string",
+					},
+				},
+				KeyVaultSecretDefinitions: []scaffold.KeyVaultSecretDefinition{
+					{
+						SecretName:  "storage-account-connection-string",
+						KeyVaultUrl: "${keyVault.outputs.uri}secrets/STORAGE-ACCOUNT-CONNECTION-STRING",
+					},
+				},
+			}, nil
+		default:
+			return scaffold.EnvironmentVariableInformation{}, unsupportedAuthTypeError(resourceType, authType)
+		}
+	case ResourceTypeOpenAiModel: // Keep this as code template
+		switch authType {
+		case internal.AuthTypeUnspecified:
+			return scaffold.EnvironmentVariableInformation{
+				StringEnvironmentVariables: []scaffold.StringEnvironmentVariable{
+					{
+						Name:  "AZURE_OPENAI_ENDPOINT",
+						Value: "${account.outputs.endpoint}",
+					},
+				},
+			}, nil
+		default:
+			return scaffold.EnvironmentVariableInformation{}, unsupportedAuthTypeError(resourceType, authType)
+		}
+		//case OtherType: // Keep this as code template
+		//	switch authType {
+		//	default:
+		//		return scaffold.EnvironmentVariableInformation{}, unsupportedAuthTypeError(resourceType, authType)
+		//	}
+	case ResourceTypeHostContainerApp: // todo improve this and delete Frontend and Backend in scaffold.ServiceSpec
+		switch authType {
+		case internal.AuthTypeUnspecified:
+			return scaffold.EnvironmentVariableInformation{}, nil
+		default:
+			return scaffold.EnvironmentVariableInformation{}, unsupportedAuthTypeError(resourceType, authType)
+		}
+	default:
+		return scaffold.EnvironmentVariableInformation{}, unsupportedResourceTypeError(resourceType)
```

Comment:
Same here: The error can help to do the check work when added new resource type.